src/mtcp/mtcp_restart.c

```diff
@@ -1262,10 +1273,32 @@ static void lock_file(int fd, char* name, short l_type)
   }
 }
 
+static char* fix_parent_directories(char* filename) {
+  int mtcp_sys_errno;
+  int i;
+  for (i=1; filename[i] != '\0'; i++) {
+    if (filename[i] == '/') {
+      int rc;
+      filename[i] = '\0';
+      rc = mtcp_sys_mkdir(filename, S_IRWXU);
+      if (rc < 0 && mtcp_sys_errno != EEXIST ) {
+        MTCP_PRINTF("error %d re-creating directory %s\n",
+		    mtcp_sys_errno, filename);
+        filename[i] = '/';  // Restore original filename
+	mtcp_abort();
+      } // else parent directory already exists or was re-created
+      // FIXME:  We should unlink parent directories if we re-created them.
+      filename[i] = '/';  // Restore original filename
+    }
+  }
+  return filename;  // Success.  All parent directories now exist.
+}
+
 #if 0
 /* Adjust for new pwd, and recreate any missing subdirectories. */
 static char* fix_filename_if_new_cwd(char* filename)
 {
+  int mtcp_sys_errno;
```

Comment:
Github doesn't allow adding comment on unmodified lines so adding it here:

Lower in the function (lines 1314-1330), we have the logic to recreate directory tree. Why not replace this with the new function that you introduced?